### ISSUE TITLE
Allow default values of particle variables to be variables themselves

### ISSUE THREAD

**Issue by erikvansebille** [OceanPARCELS/parcels#101]
In the notebook tutorial, we highlight the usecase of a custom ParticleClass for calculating distance travelled. However, this currently still requires an `__init__` statement, as the `prev_lon` Variable cannot be defaulted to `lon` within the declaration of `prev_lon` (and same for `prev_lat`).

Ideally we would be able to do something like
```
class DistParticle(JITParticle):
    distance = Variable('distance', dtype=np.float32, default=0) 
    prev_lon = Variable('prev_lon', dtype=np.float32, default='lon')
    prev_lat = Variable('prev_lat', dtype=np.float32, default='lat')
```
While this is almost possible with `attrgetter` (via `default=attrgetter('lon')`), this currently fails with the error
```
/Users/erik/Codes/PARCELScode/parcels/particle.py in __init__(self, name, dtype, default)
     12         self.name = name
     13         self.dtype = dtype
---> 14         self.default = self.dtype(default)
     15 
     16     def __get__(self, instance, cls):

TypeError: float() argument must be a string or a number
```

**Comment by mlange05** [comment]
Just for the record, there is another implementation subtlety here that might affect things: For JIT the "default" always gets set at particle initialisation, so `prev_lon` will default to the **initial** value of `lon`, whereas the Scipy version will default to the value of `lon` **at the time we access** `prev_lon`. This means if `lon` changes between particle initialisation and calling `p.prev_lon` we get different behaviour between the two versions; a rare case, but good to keep an eye on.

**Comment by mlange05** [comment]
Has this been fixed sufficiently in PR #109?

**Comment by erikvansebille** [comment]
No, it seems not to have been fixed yet. I just created a branch `variable_initialisation_from_variables` showing this in practice. The example of the `'prev_lon` and `prev_lat` needed for distance calculation still can't simply be initialised to `lon` and `lat`.

However, I don't think this is really a priority now; there's a good workaround, and for almost all other usecases the initialisation process works. Defer to after v0.9?
